### ISSUE TITLE
Plugin list not changing

### ISSUE THREAD

**Issue by whyareallthegoodusernamestaken** [afragen/github-updater#420]
The list of plugins on the settings page will not update when adding and removing plugins.  I even went so far as to dump the $plugins variable in the get_plugin_meta function in the Plugin class to see if the issue was caused by some weird database caching, but it has the list in it.  
  On my other sites I am able to update my plugins just fine, but this one had a small issue from an older version of one of my plugins with the wrong bitbucket URI in the header, and it caused the system to overlook another plugin with the same URI.  Previously, when I changed the URI, your plugin would pull the correct info, this time is just pulled the same info. 

  Did you add some kind of caching?  I honestly didn't see anything, but I also didn't write your code, and could have easily overlooked something.  If it helps, I am completely up to date on WordPress, my theme (Divi), and your plugin.  I did give up, and just reinstalled everything, and the settings page isn't showing the list of installed plugins at all.

**Comment by afragen** [comment]
Any caching is only for 12 hours. If there are no errors, don't worry.

tl;dr

The plugin settings page only displays plugins or themes that are in private GitHub or Bitbucket repos. If there are no version updates more/all of the plugins will show with blank areas for the GitHub access token and unchecked Bitbucket boxes. If these are public repos don't change this.

The reason that sometimes you can see all the plugins is a short circuit I put into the code when there is no version update. This allows for not making several `wp_remote_get()` calls per plugin and makes GHU more performant. The side effect is that the call to the repo's meta data gets skipped and so the variable that describes the repo as public or private isn't present.

By default I set the variable to private so that the user will see where they need to put an access token or check a box. If there is a private repo that requires one of these and doesn't have it an error will be displayed.

**Comment by whyareallthegoodusernamestaken** [comment]
I will check back tomorrow for sure then.  Fyi, all our plugins are
private, and I forgot to mention it, the settings page showed the same 4
plugins even after I had deactivated, and deleted them through typical wp
fashion.  Thank you for your quick response.

**Comment by afragen** [comment]
To clear the cache you can use the 'Check Again' button on the update-core.php page. This might help too.

**Comment by afragen** [comment]
I think you showed me an odd bug. I think it's fixed in the latest `develop` branch. 

If using the 'Check Again' button worked, the fix should work. You can test it out by using the branch switch to go to the develop branch.

**Comment by whyareallthegoodusernamestaken** [comment]
​​Just checked, and the plugins are pulling through.  All of them were
unchecked as private repos, but I was able to check them and successfully
save the settings.  Thank you again.

Jeffrey Mark Baldridge | Custom Programmer

<http://www.google.com/url?q=http%3A%2F%2Fturnthepage-onlinemarketing.com&sa=D&sntz=1&usg=AFQjCNEgG7w2DrQmBKR3DGgxtR7AHZ-w4w>
   206 SE 2nd Street
   Lee's Summit, MO 64063
   P: (816) 527-8371 <8165278371> ext: 413
   F: *(866) 255-2740 <8662552740>*

**Comment by afragen** [comment]
Jeffrey, thanks for confirming it's now working.

How did you originally add to plugins, via the Install Plugin tab? If so, this is part of what I fixed yesterday.

If there's no longer an issue please close this one out when we're done. :smiley:

**Comment by whyareallthegoodusernamestaken** [comment]
Yes, I did install all the plugins via your installer.  Thank you again for your quick response, and all the work you did for this plugin.
